### PR TITLE
SISRP-16309 - Create Feature Flag and Back-End for CS Billing

### PR DESCRIPTION
**Pull request by davidlimcheng**
https://jira.berkeley.edu/browse/SISRP-16309

*resumes finger crossing*

I did default the feature flags to `true` in all of the `.yml` files except for `settings.yml` - not sure if this is the convention I should be following or not. Let me know if otherwise, and I'll push a new commit up with correct flags.

### CODE REVIEW

File: fixtures/xml/campus_solutions/billing_details.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    <ITEM_BALANCE type="float">-650</ITEM_BALANCE>
+    <ITEM_EFFECTIVE_DATE>2016-01-12</ITEM_EFFECTIVE_DATE>
+    <ITEM_DESCRIPTION>Room and board</ITEM_DESCRIPTION>
+    <ITEM_DUE_DATE>2006-11-15</ITEM_DUE_DATE>
+    <ITEM_REFERENCE_DESCRIPTION>CM0002815148</ITEM_REFERENCE_DESCRIPTION>
+    <ITEM_STATUS>Paid</ITEM_STATUS>
+    <ITEM_TYPE>Payment</ITEM_TYPE>
+    <ITEM_TERM_DESCRIPTION>Fall 2006</ITEM_TERM_DESCRIPTION>
+    <ITEM_TERM_ID>2168</ITEM_TERM_ID>
+    <ITEM_DUE_FLAG>DUE_NOW</ITEM_DUE_FLAG>
```
Comment (paulkerschen) [inline fixtures/xml/campus_solutions/billing_details.xml]: @davidlimcheng For readability, here and below in this fixture, could we indent all the `<ITEM....` elements nested inside `<ACTIVITY_ITEM>` one more level?

**Comment by paulkerschen** [comment]
This all looks really good to me, @davidlimcheng, and the 1-out-of-2 Travis failure looks like a fluke. 

If you don't mind fixing the one XML formatting issue and amending the commit (preferable to adding a second commit), I think it's good to merge.

File: config/settings.yml [hunk 2]
```diff
@@ -426,6 +426,7 @@ features:
   cs_advising_gte_forms: false
   cs_advising_semester_study_list: false
   cs_advising_show_advisors: false
+  cs_billing: false
```
Comment (paulkerschen) [inline config/settings.yml]: @davidlimcheng Yes, this is a good convention to follow; default to `false` and then whitelisting development and test environments to `true` makes it less likely that we'll turn it on by accident in some environment (such as production, eek).

**Comment by davidlimcheng** [comment]
Thanks @paulkerschen!  Did not know you can amend commits, that will save me a ton of time in the future.

File: fixtures/xml/campus_solutions/billing_details.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+<UC_SF_ACTIVITY>
+  <SUMMARY>
+     <AMOUNT_DUE_NOW type="float">49102</AMOUNT_DUE_NOW>
```
Comment (christianvuerings) [inline fixtures/xml/campus_solutions/billing_details.xml]: @davidlimcheng indentation looks a bit off here.

Comment (davidlimcheng) [inline fixtures/xml/campus_solutions/billing_details.xml]: @christianvuerings wow, nice catch.  Eyes of a hawk.  Fixed & pushed.

**Comment by christianvuerings** [comment]
lgtm - good to merge with 1 green travis

**Pull request merged**